### PR TITLE
Add examples how to allow browser access to Gnome extensions connector

### LINKED ISSUE #4177: Document how to enable browser access to native Gnome connector (extensions.gnome.org)
Default profile for Chrome/Firefox disables access to native Gnome connector and making extensions.gnome.org actions like listing extensions, installing, enabling etc impossible.

It would be great if profiles mentios what needs to be done in order to reenable this function for Gnome users. Here's a suggestion for Chrome:

into `chromium-common.profile`
```
# Add the next 3 lines to your chromium-common.local if you want Google Chrome/Chromium browser
# to have access to Gnome extensions (extensions.gnome.org) via browser connector
noblacklist ${PATH}/python3*
noblacklist /usr/lib/python3*
noblacklist /usr/lib64/python3*
```

And Firefox:

into `firefox.profile`:

```
# Add the next 7 lines to your firefox.local if you want Firefox to gain
# access to Gnome extensions (extensions.gnome.org) via browser connector
dbus-user.own org.gnome.ChromeGnomeShell
dbus-user.own org.gnome.Shell
dbus-user.own ca.desrt.dconf
dbus-user.own org.gtk.vfs.Daemon
noblacklist ${PATH}/python3*
noblacklist /usr/lib/python3*
noblacklist /usr/lib64/python3*
```

I have tested this on Arch Linux x86_64, rolling and current master.

### PR DESCRIPTION
**Pull request by jose1711**
Fixes #4177.

### CODE REVIEW

File: etc/profile-a-l/chromium-common.profile [hunk 1]
```diff
@@ -30,6 +30,10 @@ include whitelist-runuser-common.inc
 include whitelist-usr-share-common.inc
 include whitelist-var-common.inc
 
+# Add the next line to your chromium-common.local if you want Google Chrome/Chromium browser
+# to have access to Gnome extensions (extensions.gnome.org) via browser connector
+#include allow-python3.inc
```
Comment (rusty-snake) [inline etc/profile-a-l/chromium-common.profile:35]: Move it up to line 14:

```
noblacklist ${HOME}/.local/share/pki

# Add the next line to your chromium-common.local if you want chromium-based browser
# to have access to Gnome extensions (extensions.gnome.org) via browser connector
#include allow-python3.inc

include disable-common.inc
```

File: etc/profile-a-l/firefox.profile [hunk 2]
```diff
@@ -14,6 +14,14 @@ mkdir ${HOME}/.mozilla
 whitelist ${HOME}/.cache/mozilla/firefox
 whitelist ${HOME}/.mozilla
 
+# Add the next 5 lines to your firefox.local if you want Firefox to gain
+# access to Gnome extensions (extensions.gnome.org) via browser connector
+#dbus-user.talk org.gnome.ChromeGnomeShell
+#dbus-user.talk org.gnome.Shell
+#dbus-user.talk ca.desrt.dconf
+#dbus-user.talk org.gtk.vfs.Daemon
+#include allow-python3.inc
```
Comment (rusty-snake) [inline etc/profile-a-l/firefox.profile:23]: Same, and I would just say "the next/following lines". Including a number means that if someone add a new line (e.g. `dbus-user.talk org.gnome.ChromeGnomeShell2`) this must be updated too to be correct.

**Review by rusty-snake — COMMENTED** [review]
:+1:

**Review by rusty-snake — CHANGES_REQUESTED** [review]
#### chromium-common.profile

- Keep the accidental removed `#include chromium-common-hardened.inc.profile`
- FWIW: this common is used by all chromium based (e.g. brave or vivaldi).

#### firefox.profile

We have already a GNOME Shell integration snippet in firefox-common-addons.profile (which is the proper profile for that, if I think again):

https://github.com/netblue30/firejail/blob/83b95efe1a8f5b4386243a50d28065b14567faf3/etc/profile-a-l/firefox-common-addons.profile#L73-L79

So we can harden this one to keep the system-bus blocked and use filter instead of allow for the session-bus.

**Comment by jose1711** [comment]
> #### chromium-common.profile
> 
>     * Keep the accidental removed `#include chromium-common-hardened.inc.profile`

Oops, note taken down.

>     * FWIW: this common is used by all chromium based (e.g. brave or vivaldi).
> 
> 
> #### firefox.profile
> 
> We have already a GNOME Shell integration snippet in firefox-common-addons.profile (which is the proper profile for that, if I think again):
> 
> https://github.com/netblue30/firejail/blob/83b95efe1a8f5b4386243a50d28065b14567faf3/etc/profile-a-l/firefox-common-addons.profile#L73-L79

Hmm.. tested with `firefox-common-addons.profile` included but it stays broken for me. I had to add the following 3 lines to get rid of the errors:
```
dbus-user.talk org.gnome.ChromeGnomeShell
dbus-user.talk org.gnome.Shell
dbus-user.talk ca.desrt.dconf
```

Wonder if anyone else can confirm the same with (other than) Arch Linux too.


> 
> So we can harden this one to keep the system-bus blocked and use filter instead of allow for the session-bus.

**Comment by rusty-snake** [comment]
> Hmm.. tested with firefox-common-addons.profile included but it stays broken for me. I had to add the following 3 lines to get rid of the errors:

Makes sense because firefox uses `dbus-user filter`. Maybe

```
# GNOME Shell integration (chrome-gnome-shell) needs dbus and python 
noblacklist ${HOME}/.local/share/gnome-shell 
whitelist ${HOME}/.local/share/gnome-shell 
dbus-user.talk ca.desrt.dconf
dbus-user.talk org.gnome.ChromeGnomeShell
dbus-user.talk org.gnome.Shell
# Allow python (blacklisted by disable-interpreters.inc) 
include allow-python3.inc 
```

**Comment by jose1711** [comment]
> ```
> # GNOME Shell integration (chrome-gnome-shell) needs dbus and python 
> noblacklist ${HOME}/.local/share/gnome-shell 
> whitelist ${HOME}/.local/share/gnome-shell 
> dbus-user.talk ca.desrt.dconf
> dbus-user.talk org.gnome.ChromeGnomeShell
> dbus-user.talk org.gnome.Shell
> # Allow python (blacklisted by disable-interpreters.inc) 
> include allow-python3.inc 
> ```

Worked for me, made adjustments and force-pushed now

**Review by rusty-snake — APPROVED** [review]

**Comment by glitsj16** [comment]
I haven't used chrome-gnome-shell in a while, but I recall it also used to show desktop notifications informing the user of update availability. If we allow GNOME Shell integration this way, shouldn't it make sense to add a filter for notifiers too, something like the below?
```
dbus-user.talk org.freedesktop.Notifications
dbus-user.talk org.kde.knotify
dbus-user.talk org.kde.StatusNotifierWatcher
```

**Comment by rusty-snake** [comment]
> dbus-user.talk org.freedesktop.Notifications

Make sense if it's used.

> dbus-user.talk org.kde.knotify

Is this used? It's GNOME.

> dbus-user.talk org.kde.StatusNotifierWatcher

Tray icons are dead on GNOME.

**Comment by glitsj16** [comment]
> Is this used? It's GNOME.
    dbus-user.talk org.kde.StatusNotifierWatcher

Correct, my bad.

> Tray icons are dead on GNOME.

That's exactly why people who want them install GS extensions. Besides, there are other options for controlling GS extensions now in GNOME 40. And installing those via native package management tools is always going to be safer anyway IMO.

**Comment by jose1711** [comment]
> I haven't used chrome-gnome-shell in a while, but I recall it also used to show desktop notifications informing the user of update availability. If we allow GNOME Shell integration this way, shouldn't it make sense to add a filter for notifiers too, something like the below?
> 
> ```
> dbus-user.talk org.freedesktop.Notifications
> dbus-user.talk org.kde.knotify
> dbus-user.talk org.kde.StatusNotifierWatcher
> ```

No idea - I never enable browser notifications. Maybe someone who really needs them can comment and prepare another commit.

**Comment by rusty-snake** [comment]
Merged, thanks.

**Pull request merged**